.github/actions/setup-project/action.yml

```diff
@@ -0,0 +1,19 @@
+name: Setup project
+description: Install required tools
+inputs:
+  node-version:
+    required: false
+    description: node version to install
+    default: 20.10.0 # is already cached in Ubuntu 22.04 runner
+runs:
+  using: composite
+  steps:
+    - name: Enable corepack
+      shell: bash
+      run: |
+        corepack enable
```

Comment:
Won't this require devs to also enable corepack?

Reply:
I do enable corepack to get yarn 4 installed for GH action builds

Reply:
https://yarnpkg.com/getting-started/install

Reply:
I meant locally.

Reply:
I don't think so - I haven't had to do it, but had to do it on CI, so hard to tell

Reply:
Doesn't look like it:
```
% yarn
error This project's package.json defines "packageManager": "yarn@4.1.1". However the current global version of Yarn is 1.22.21.

Presence of the "packageManager" field indicates that the project is meant to be used with Corepack, a tool included by default with all official Node.js distributions starting from 16.9 and 14.19.
Corepack must currently be enabled by running corepack enable in your terminal. For more information, check out https://yarnpkg.com/corepack.
```

Reply:
so I guess it's same case for yarn@3 then? So it seems corepack needs to be enabled. Is there any drawback to that?

Reply:
Unless it's enabled by default, I don't think we should require it.

Reply:
I mean, this is for our root project setting, not workspaces - those would still use whatever the project is using. Are you concerned about contributors?

Reply:
I don't think it's something you want to turn off - from node page:
>  will identify whatever package manager is configured for the current project, download it if needed, and finally run it.
So it's a package manager for package managers really - yarn 4 being part of it.